WordPress/Classes/Utility/Analytics/WPAppAnalytics.m

```diff
@@ -0,0 +1,202 @@
+//
+//  WPAppAnalytics.m
+//  WordPress
+//
+//  Created by Diego E. Rey Mendez on 3/16/15.
+//  Copyright (c) 2015 WordPress. All rights reserved.
+//
+
+#import "WPAppAnalytics.h"
+
+#import "WPAnalyticsTrackerMixpanel.h"
+#import "WPAnalyticsTrackerWPCom.h"
+#import "WPTabBarController.h"
+#import "WordPressComApiCredentials.h"
+
+NSString* const WPAppAnalyticsDefaultsKeyUsageTracking = @"usage_tracking_enabled";
+static NSString* const WPAppAnalyticsKeyLastVisibleScreen = @"last_visible_screen";
+static NSString* const WPAppAnalyticsKeyTimeInApp = @"time_in_app";
+
+@interface WPAppAnalytics ()
+
+/**
+ *  @brief      Timestamp of the app's opening time.
+ */
+@property (nonatomic, strong, readwrite) NSDate* applicationOpenedTime;
+
+/**
+ *  @brief      If set, this block will be called whenever this object needs to know what the last
+ *              visible screen was, for tracking purposes.
+ */
+@property (nonatomic, copy, readwrite) WPAppAnalyticsLastVisibleScreenCallback lastVisibleScreenCallback;
+@end
+
+@implementation WPAppAnalytics
+
+#pragma mark - Dealloc
+
+- (void)dealloc
+{
+    [self stopObservingNotifications];
+}
+
+#pragma mark - Init
+
+- (instancetype)init
+{
+    [self doesNotRecognizeSelector:_cmd];
```

Comment:
:+1:

The idea is to not let any space for doubt.  If the parent code tries to init an instance through this method, I want it to fail and make noise! :)